### ISSUE TITLE
Let html tag in markdown be foldable

### ISSUE THREAD

**Issue by SHF101202021** [yzhang-gh/vscode-markdown#1485]
### Proposal
While writing massive texts, folding is a very good function to be focus the current content.
For example, long programming texts in the article (Please look at the `line 8`).

![image](https://github.com/user-attachments/assets/aaa8a0c8-c920-4c6a-87ce-79e835958a61)
![image](https://github.com/user-attachments/assets/a9fc18b8-ef09-4051-bcc1-0c5b0077bd44)

However, when I want to fold some html tags like `<details>`, it fail (Please look at the `line 4`).

![image](https://github.com/user-attachments/assets/154cba94-c716-4e0b-99b3-b667851604a6)
![image](https://github.com/user-attachments/assets/27c72163-8a54-484c-98eb-3cebca85ec47)

Basically, it just fold elements which are sequentially arranged. 
Hence, I want to request to consider the html tags in formatting process.

**Comment by o-l-a-v** [comment]
Wouldn't this be a VSCode markdown feature? 🤔

Found a workaround using HTML comments from here: <https://stackoverflow.com/questions/3521826/is-there-region-code-for-html>.

````markdown
<!-- #region Region 1 -->

<details>
<summary>Test</details>

```pwsh
Write-Output -InputObject 'Test'
```

</details>

<!-- #endregion -->
````

![Image](https://github.com/user-attachments/assets/d3e54003-d8a9-4e6a-9a36-646cfbed50c9)

---

Edit: Made feature request in the VSCode repo: <https://github.com/microsoft/vscode/issues/249613>.
